sklearn_pandas/dataframe_mapper.py

```diff
@@ -47,6 +47,8 @@ def _get_feature_names(estimator):
     """
     if hasattr(estimator, 'classes_'):
         return estimator.classes_
+    elif hasattr(estimator, 'get_feature_names_out'):
```

Comment:
@falcaopetri wondering if someway we can make this backward compatible. One option is using an additional DataFrameMapper function that explicit lists functions to use. For instance `mapper = DataFrameMapper(..., col_name_func=['classes_', 'get_feature_names_out', 'get_feature_names'])`. The disadvantage is that users have to know about these functions. What do you think ?

Reply:
@ragrawal I'm not sure what you mean by backward compatible. In some sense, it is backward compatible since we would still fallback to `get_feature_names`. But I agree that there is a change in the behavior for specific estimators when using sklearn <1.0 vs >=1.0 (it happens to those that did not have `get_feature_names` in sklearn 0.23 but have `get_feature_names_out` in >=1.0, such as `StandardScaler` and `SelectKBest`).

Reply:
My biggest concern is it will break someone's model in production if the libraries are not pinned. I am thinking of what will be a more graceful approach where we don't break someone's existing model in production. If there is no graceful way I am open to accepting the above PR as it does clean up names.

Reply:
Agreed. The only approach I see is to check which version of `sklearn` is installed. If user's model has unpinned `sklearn` or `sklearn-pandas` and depends on current behavior of `transformed_names_`, things will break one way or another. Anyway, we could even decide to delay the PR till it's really needed (`sklearn>=1.2`).

There is of course the possibility to forcibly keep the current behavior. For example, we could keep a `denylist` with all estimators that had their behavior changed (given the criteria in my last comment). We would ignore `get_feature_names_out` for them. This assumes that `sklearn` will keep all other `get_feature_names_out` compatible with `get_feature_names`.